src/libsyntax/ext/deriving/bounds.rs

```diff
@@ -51,7 +51,8 @@ pub fn expand_deriving_bound<F>(cx: &mut ExtCtxt,
         path: Path::new(vec!("std", "marker", name)),
         additional_bounds: Vec::new(),
         generics: LifetimeBounds::empty(),
-        methods: vec!()
+        methods: Vec::new(),
+        associated_types: Vec::new(),
```

Comment:
either way it is now more consistent (used to be both `Vec:;new()` and `vec!()` here)